### DISCUSSION TITLE
Does Stellarium work with Synscan WiFi ON LINUX?

### DISCUSSION THREAD

**Discussion by Twilight-Logic** [Stellarium/stellarium#4010] [General]
Keep finding instructions for Windows, but would like to use Stellarium with my HEQ5Pro mount via a Synscan WiFi dongle. First need to confirm that the dongle would work on Linux please? Then would like to know where to find instructions to set it up? Will keep looking in the meantime.

**Comment by gzotti** [comment]
If you send me a SynScan HEQ5Pro mount and dongle to keep, I can try it out for you. :-) Or maybe you ask at SynScan whether their hardware works with Linux. Does the dongle need dedicated drivers with specialized functionality or is it just a generic WiFi dongle?

**Reply by Twilight-Logic** [reply]
Don't know as I haven't purchased it yet. Given the price it is probably generic. I imagine that being a WiFi dongle it would have to communicate over IP. Apparently it supports ASCOM tools from a tablet or phone, but on a PC ASCOM is supported only on Windows. I did find something called INDI which is supported by Stellarium, but need to do more research on it.

**Comment by Dequinzr** [comment]
Bonjour,
J'utilise stellarium pour guider mon Dobson Goto 254 muni d'un Wifi intégré et ce en utilisant le driver ASCOM configuré pour utiliser Synscan pro sur mon laptop en Windows et l'appli Synscan Pro sur mon GSM. Si ça vous intéresse, communiquez-moi votre adresse e-mail et je vous enverrai un document Word que j'ai écrit pour expliquer comment tout configurer.

**Comment by Dequinzr** [comment]
Le principe c'est de connecter le laptop et le GSM au Wi-fi du télescope, de lancer Synscan Pro sur le GSM, le connecter au télé puis lancer Stellarium, et lancer le gestionnaire de guidage.
Stellarium envoi les coordonnées de l'objet de votre choix vers Synscan qui pointe l'instrument.

**Comment by Twilight-Logic** [comment]
Merci beaucoup. Mon Français est très faible et j'avais beaucoup oubliée, mais j'avis compri. Un question: quest que c'est GSM? L'explication de ton configuration m'interesse. Mon adresse e-mail est subs@qcontinuum.plus.com. S'il vous plait, enverrai moi ce document. Je serias reconassant de le recevoir. Sur Linux, il faut que j'ultilise INDI. Alors, je ne suis pas certain que ca fonctionne de la meme façon que ASCOM, mais peut etre votre document ma donne quelqe idé. ASCOM ces't seulment pour Windows? J'avis trouvé aussi ASCOM Alpaca, mais je ne pas le connais, alors il fault que je fait les recherches.
Je vous remerci.

**Comment by Dequinzr** [comment]
Hi, no problem to communicate in English. I will send you my document (It is written in English). I am simply making some changes into it according to new information I have found.  GSM = Mobile phone ;-)

**Comment by Twilight-Logic** [comment]
Regarding GSM, that was what I related it to, just wasn't sure. The Synscan WiFi documentation does say it works with an 'app' on a tablet or phone, so that makes sense. I am trying to figure out whether I can use it from my Linux PC. I made some progress with INI and Kstars via USB, but would be nice to be able to use it wirelessly. It might help to know how the connection is made but I suspect I might end up having to use something like a Raspberry Pi with INDI connected to WiFi. Thank you sending me your setup document.

As more of a general update to the enquiry, much of the documentation I have encountered seems to assume prior knowledge. For example, in the two instruction documents that I have found so far, there are instructions to install (and in one compile) various packages on Linux, but it then, without explanation they jump to an instruction to go to a tab or configuration option in a GUI when the installed tools appear to be command line based? With a bit more research I eventually discovered that calling up the configuration GUI is done inside an application such as Kstars or Stellarium when one configures INDI. I have now made some progress with Kstars, but not Stellarium which just tells me that nothing is connected. Still working on it! I suspect that perhaps the Stellarium plugin, unlike Kstars that can launch the INDI server with the aplication, requires, not just the INDI server to be lauched from the command line and running, but also the mount to be physically connected. Unfortunately, I don't have the appropriate interface hardware (USB, WiFi or otherwise) for it yet. I don't want to spend 100 pounds/euro on a WiFi dongle only to find out that it is not supported and does not work. This was the reason for my initial question. I figured someone must have had a go at doing this before?

**Comment by Dequinzr** [comment]
I have no knowledge in Linux (sorry) but i am familiar with Windows platform. I have worked about 1 year to find out how to use a planetarium software to guide my telescope. Now I am using both Stellarium and Guide9 in connection with the mobile phone to manage my Dobson Skywatcher telescope. Please find the configuration document attached.
[Guiding Skywatcher with Stellarium V5 (Robert and Benoit)-EN.pdf](https://github.com/user-attachments/files/18164543/Guiding.Skywatcher.with.Stellarium.V5.Robert.and.Benoit.-EN.pdf)

**Reply by Twilight-Logic** [reply]
> I have worked about 1 year to find out how to use a planetarium software to guide my telescope.

Wow! What chance have I got in one day or even one week....! :-)
Thank you for the link!

**Comment by gzotti** [comment]
From our User Guide:

> ### INDI settings
> (see also 16.2.7) You will need access to an INDI server through a TCP/IP connection. Please see
> INDI documentation for details. You will need to select the correct INDI device to manage it.

So yes, you need to setup INDI and have the server with your instruments running before connecting from Stellarium. Whether the INDI server is running on the same Linux PC as Stellarium or on a separate Raspberry Pi or whatever device is up to you.

**Reply by Twilight-Logic** [reply]
Ah, ok so I guess that confirms it. Kstars manages to bring up the GUI for configuration purposes without the mount being physically connected, just with the server running. Evidently Stellarium works a little differently. I will now focus on setting up a USB connection.

**Comment by Dequinzr** [comment]
My pleasure to share 😁

Le mar. 17 déc. 2024, 12:56, Twilight-Logic ***@***.***> a
écrit :

**Reply by Twilight-Logic** [reply]
Received. Thank you.

**Comment by Twilight-Logic** [comment]
So it turns out that my Synscan handset has a USB A type port. According to the image in the V5 manual, its a V5 handset. The image in the V4 manual does not show this. According to the V5 manual, this can be used to "communicate with a computer via RS-232". When I plug this in, it comes up as a "Prolific Technology, Inc. USB-Serial Controller" device and I get a /dev/ttyUSB0 port. Perhaps I don't need a USB adapter after all? However, here is where it get a bit confusing. This is from the same manual: "The application running on the PC controls the motor controller directly. Currently, the PC direct mode is used to update the motor controller's firmware.". Ok, so does it control the telescope, or is it just a means to update the firmware? Or both? I connected it to Stellarium and, after setting the handset to "PC Direct Mode" tried both the serial method and the INDI method, but unfortunately neither seemed to work.

In Kstars at least I get this feedback:

> 2024-12-17T13:32:31: [ERROR] Serial read error: Timeout error. 

I tried all of the baud rates, but get the same response, which is disappointing. I tried a manual connection using PuTTy. This connected a session. but there was no response to any input. Perhaps I need an adapter after all?

Obviously, this is a hardware issue and until I get it resolved, I will not be able to get Stellarium to work, so for the moment, I am looking elsewhere for answers.

**Reply by gzotti** [reply]
IIRC the Synscan mount can speak LX200 to Stellarium directly over the USB/serial connection.

**Reply by Twilight-Logic** [reply]
Interesting. I can pick a planet or the sun on the handset and it will slew to the position and then go back to parked on request as expected which would suggest that the motor drive is working and the mount is slewing normally. It sounds like I should be getting a response from that USB COM port.

**Reply by gzotti** [reply]
On Linux, you may need permissions to access the serial port. Are you in the right user group? (dialout, tty, ... depends on distribution)

**Reply by Twilight-Logic** [reply]
Fair point, but yes, I did check. My username is in the dialout group and I can use other serial devices. However, you have got me wondering whether it also needs UDEV rules? Arduinos and other boards do seem to need them.

**Reply by gzotti** [reply]
Cannot answer this as of today. If you are in the dialout group, you should be able to use some tty program and connect to the controller. If not, it's a system config issue.

**Reply by Twilight-Logic** [reply]
Thanks. I appreciate the replies. I did ask on Linux IRC forum about USB devices and dialout. The device is group owned by 'dialout' and my username is a member of that group and it does have rw privileges, so the upshot is that this ought to suffice and UDEV rules should not be required. I think the next step is as you suggest, to establish and verify communication between the system and handset. At the moment I get a connection but no responses, however I know nothing about the Synscan/LX200 protocol, so I need to research that.

**Comment by Twilight-Logic** [comment]
Made a lot of progress this morning. It turns out that I was using the wrong INDI driver. When using the USB connector on the V5 handset, the USB-to-serial converter built into the handset can act like an EQDIR cable, bypassing the handset in the process. However, for this to work, the handset must be set to "PC_Direct_Mode", which enables the mount to be driven directly using an EQMod driver. Also, the driver to use when starting the INDI server is indi_eqmod_telescope, NOT indi_synscan_telescope. It should be noted, that the handset will also continue to operate the mount, but an EQMOD (EQDIR - USB to RJ45) cable could also be used instead to completely remove the need for the handset. Evidently it the EQMOD driver that uses the protocols required to directly drive the motor drive controller on the mount - not the Synscan one. That being established and the solution confirmed to be working, I think this can be closed.
